crates/libs/bindgen/src/types/cpp_method.rs

```diff
@@ -825,7 +825,7 @@ fn is_param_retval(ty: &Type, param: Param, hint: ParamHint) -> bool {
     {
         return false;
     }
-    if hint.is_array() {
+    if ParamHint::from_param(param).is_array() {
```

Comment:
```suggestion
    // This is reevaluated to detect unsupported array parameters.
    // https://github.com/microsoft/windows-rs/issues/3384
    if ParamHint::from_param(param).is_array() {
```

Reply:
Ideally we'd have a formal test beyond code gen but at least this will give future maintainers a hint that something is happening here.